backend/app/Models/Transaction.php

```diff
@@ -10,11 +11,17 @@ class Transaction extends Model
 
     public function transactionCard()
     {
-        return $this->hasOne(App\Models\Card::class, 'id', 'card_id');
+        return $this->hasOne(Card::class, 'id', 'card_id');
     }
 
     public function transactionUser()
     {
         return $this->hasOne('App\Models\User');
     }
+
+    public function getPriceAttribute($value)
+    {
+        //price is stores as a bigint in the db, need to make it a properly formatted float.
+        return EthereumConverter::convertETHPriceToInt($value);
```

Comment:
This should be
```
return EthereumConverter::convertETHPriceToFloat($value);
```
This function will automatically be called by laravel whenever you try to access the price attribute of the transaction model. That way it is converted to a float whenever we need it.